tst/EnergyPlus/unit/ReportCoilSelection.unit.cc

```diff
@@ -0,0 +1,427 @@
+	EXPECT_EQ( RatedCoilOutDb, c1->ratedCoilOutDb );
+	EXPECT_EQ( RatedCoilOutWb, c1->ratedCoilOutWb );
+	EXPECT_EQ( RatedCoilOutHumRat, c1->ratedCoilOutHumRat );
+	EXPECT_EQ( -999.0, c1->ratedCoilOutEnth );
+	EXPECT_EQ( RatedCoilEff, c1->ratedCoilEff );
+	EXPECT_EQ( RatedCoilBpFactor, c1->ratedCoilBpFactor );
+	EXPECT_EQ( RatedCoilOadbRef, c1->ratedCoilOadbRef );
+	EXPECT_EQ( RatedCoilOawbRef, c1->ratedCoilOawbRef );
+
+	// again with setting coil inlet/outlet conditions
+	RatedCoilInDb = 23.0;
+	RatedCoilInHumRat = 0.008;
+	RatedCoilOutDb = 40.0;
+	RatedCoilOutHumRat = 0.009;
+	coilSelectionReportObj->setRatedCoilConditions( coil1Name, coil1Type, RatedCoilTotCap, RatedCoilSensCap, RatedAirMassFlow, RatedCoilInDb, RatedCoilInHumRat, RatedCoilInWb, 
+		RatedCoilOutDb, RatedCoilOutHumRat, RatedCoilOutWb, RatedCoilOadbRef, RatedCoilOawbRef, RatedCoilBpFactor, RatedCoilEff );
+	EXPECT_EQ( RatedCoilInDb, c1->ratedCoilInDb );
+	EXPECT_EQ( RatedCoilInHumRat, c1->ratedCoilInHumRat );
+	EXPECT_NEAR( 43460.9, c1->ratedCoilInEnth, 0.1 );
+	EXPECT_EQ( RatedCoilOutDb, c1->ratedCoilOutDb );
+	EXPECT_EQ( RatedCoilOutHumRat, c1->ratedCoilOutHumRat );
+	EXPECT_NEAR( 63371.3, c1->ratedCoilOutEnth, 0.1 );
+
+	Real64 entAirDryBulbTemp = 24.0;
+	coilSelectionReportObj->setCoilEntAirTemp( coil1Name, coil1Type, entAirDryBulbTemp, curSysNum, curZoneEqNum );
+	EXPECT_EQ( entAirDryBulbTemp, c1->coilDesEntTemp );
+	EXPECT_EQ( curSysNum, c1->airloopNum );
+	EXPECT_EQ( curZoneEqNum, c1->zoneEqNum );
+
+	Real64 entAirHumRat = 0.004;
+	coilSelectionReportObj->setCoilEntAirHumRat( coil1Name, coil1Type, entAirHumRat );
+	EXPECT_EQ( entAirHumRat, c1->coilDesEntHumRat );
+
+	Real64 entWaterTemp = 60.0;
+	coilSelectionReportObj->setCoilEntWaterTemp( coil1Name, coil1Type, entWaterTemp );
+	EXPECT_EQ( entWaterTemp, c1->coilDesWaterEntTemp );
+
+	Real64 lvgWaterTemp = 50.0;
+	coilSelectionReportObj->setCoilLvgWaterTemp( coil1Name, coil1Type, lvgWaterTemp );
+	EXPECT_EQ( lvgWaterTemp, c1->coilDesWaterLvgTemp );
+
+	Real64 CoilWaterDeltaT = 50.0;
+	coilSelectionReportObj->setCoilWaterDeltaT( coil1Name, coil1Type, CoilWaterDeltaT );
+	EXPECT_EQ( CoilWaterDeltaT, c1->coilDesWaterTempDiff );
+
+	Real64 lvgAirDryBulbTemp = 12.0;
+	coilSelectionReportObj->setCoilLvgAirTemp( coil1Name, coil1Type, lvgAirDryBulbTemp );
+	EXPECT_EQ( lvgAirDryBulbTemp, c1->coilDesLvgTemp );
+
+	Real64 lvgAirHumRat = 0.006;
+	coilSelectionReportObj->setCoilLvgAirHumRat( coil1Name, coil1Type, lvgAirHumRat );
+	EXPECT_EQ( lvgAirHumRat, c1->coilDesLvgHumRat );
+
+	int zoneNum = 1;
+	Real64 zoneCoolingLatentLoad = 1234.0;
+	coilSelectionReportObj->setZoneLatentLoadCoolingIdealPeak( zoneNum, zoneCoolingLatentLoad );
+	// Expect zero because it's a heating coil
+	EXPECT_EQ( 0.0, c1->rmLatentAtPeak );
+
+	Real64 zoneHeatingLatentLoad = 4321.0;
+	coilSelectionReportObj->setZoneLatentLoadHeatingIdealPeak( zoneNum, zoneHeatingLatentLoad );
+	// Expect zero because doZoneEqSetup isn't currently executed
```

Comment:
@EnergyArchmage I noticed that all calls to coilSelectionReportObj->doZoneEqSetup are currently commented out. Is that your intent at this point?